app/assets/javascripts/3dbio_viewer/src/domain/entities/UploadData.ts

```diff
@@ -0,0 +1,121 @@
+export interface UploadData {
+    dataUrl: string;
+    chains: Chains;
+}
+
+export interface Chains {
+    A: ChainObject[];
+    B: ChainObject[];
+    C: ChainObject[];
+}
+
+export interface ChainObject {
+    acc: string;
+    title: Title;
+    evalue: string;
+    cov: string;
+    start: string;
+    end: string;
+    db: string;
+}
+
```

Comment:
Some clean-architecture notes heres. That interface describes the response of the API, but we don't work directly with that from the domain. We should move all of this to a file within data/repositories, so only the repository uses it. And then we create a entity in our domain, with the fields and structure we like. For example, you have to do some transformation to the rows in the react component, add an ID, flatten the structure, and so on. Our model can be already React-friendly. This transformation from the api response to the entity is done by the repository.